src/voodoo/spellbook/spells/haste.ts

```diff
@@ -42,8 +42,9 @@ export const haste: SpellFunction = async (voodoo, accountId, upgradeConfigs) =>
 
   for (const playerId of playerIds) {
     const baseSpeed = await voodoo.getPlayerCheckStatBase({ accountId: playerId, stat: 'speed' });
+    const currentSpeed = await voodoo.getPlayerCheckStatCurrent({accountId: playerId, stat: 'speed' });
 
-    if (baseSpeed) {
+    if (baseSpeed == currentSpeed) {
       const buffedSpeed = baseSpeed * multiplier;
```

Comment:
Shouldn't call this the "buffed speed" anymore if it only represents an addition, not the new absolute speed stat.

```suggestion
      const speedBuff = baseSpeed * bonus;
```

There is more code below that is invalidated by these suggestions that you'd have to change for this PR.